src/extension/tsconfigHelper.ts

```diff
@@ -51,14 +51,12 @@ export class TsConfigHelper {
             .then(function(tsConfigJson: any): Q.Promise<void> {
                 tsConfigJson.compilerOptions = tsConfigJson.compilerOptions || {};
 
-                // Return if the setting is already correctly set.
-                if (tsConfigJson.compilerOptions.allowJs === enabled) {
-                    return Q.resolve<void>(void 0);
+                if (tsConfigJson.compilerOptions.allowJs === undefined) {
```

Comment:
I'm not a fan of using undefined explicitly. I'd consider using tsConfigJson.compilerOptions.hasOwnProperty("allowJs") or "allowJs" in tsConfigJson.compilerOptions